### ISSUE TITLE
Add anonymous access option to example config

### ISSUE THREAD

**Issue by jashapiro** [AlexsLemonade/scpca-nf#205]
We got the following error from an external user trying to run the workflow:

```
WARN: Unable to get file attributes file: s3://scpca-references/homo_sapiens/ensembl-104/salmon_index/Homo_sapiens.GRCh38.104.spliced_intron.txome -- Cause: com.amazonaws.services.s3.model.AmazonS3Exception: Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: 7MWP6C65SWQRE0EY; S3 Extended Request ID: 8bMJbK4WWfCVaHAZX1/7374llKgJA43DhUOyVhMNiPjmAuBrB0wRXwvXwjYLc3OMFX+V16/4vrk=; Proxy: null), S3 Extended Request ID: 8bMJbK4WWfCVaHAZX1/7374llKgJA43DhUOyVhMNiPjmAuBrB0wRXwvXwjYLc3OMFX+V16/4vrk=
WARN: Unable to get file attributes file: s3://scpca-references/homo_sapiens/ensembl-104/salmon_index/Homo_sapiens.GRCh38.104.spliced_intron.txome -- Cause: com.amazonaws.services.s3.model.AmazonS3Exception: Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: 7MWJ98K1K52VB4DX; S3 Extended Request ID: GpkokV/BYtc9E+vE7kOF02mSTYzKHhXesV6pqXa0kWb/1bgF2HWSlEnhFxRF6SEzq/5rIa455qc=; Proxy: null), S3 Extended Request ID: GpkokV/BYtc9E+vE7kOF02mSTYzKHhXesV6pqXa0kWb/1bgF2HWSlEnhFxRF6SEzq/5rIa455qc=
Error executing process > 'map_quant_rna:alevin_rad (run01-rna)'

Caused by:
 Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: 7MWMZ20NE9MVSS1V; S3 Extended Request ID: Ii4rb5EhpLCkx9Ko/15jf8ezigizUqoQhDFstbm3pepVUekAoeFh5rpac+w40NLEpHysibAoSo4=; Proxy: null)
```

This looks like a permissions issue with S3, but when I checked, all of the relevant files are public, and in theory the bucket is set to allow listing, so I am not at this point sure what is going on. 

Some other thoughts that may be relevant:
- the "file" in question is actually a directory. This has not caused trouble in the past, but I could imagine that it might be a potential problem. If that is the issue, we can modify the workflow to get all of the relevant files, but this will require more testing.
  - I note that this problem seemed to come up after #204 may have been partially solved, so the directory question seems a likely culprit. 
- It may be that AWS credentials need to be set up for the user, even if the files are public. We will need to ask about what credentials may be set.

As the next step in debugging, I asked to see what happens if the user tries `aws s3 ls s3://scpca-references/homo_sapiens/ensembl-104/salmon_index/Homo_sapiens.GRCh38.104.spliced_intron.txome` to see if we get any more relevant data there from a direct request for the file. Will update this issue if/when we get a response there

**Comment by jashapiro** [comment]
I asked the user to do some testing, and interestingly the following worked as expected:
```
$ aws s3 ls s3://scpca-references/homo_sapiens/ensembl-104/salmon_index/Homo_sapiens.GRCh38.104.spliced_intron.txome --no-sign-request
              PRE Homo_sapiens.GRCh38.104.spliced_intron.txome/
```

However, this did not:
```
$ aws s3 ls s3://scpca-references/homo_sapiens/ensembl-104/salmon_index/Homo_sapiens.GRCh38.104.spliced_intron.txome

An error occurred (AccessDenied) when calling the ListObjectsV2 operation: Access Denied
```

In other words, a completely anonymous request was able to get a listing, but when sending credentials, access was denied. This does not look like a bad credentials issue, as I think those get a different issue.

I tried adding list permissions to the `Authenticated users group (anyone with an AWS account)` as well, but that doesn't seem to have made any difference (it was a shot in the dark).

(Side note: I also checked (on my own) that adding `--recursive` to a request with `--no-sign-request` worked, which is what will be needed later.)

If anyone with more AWS experience can have a look and maybe provide some thoughts, that would be appreciated. Tagging @arkid15r and @davidsmejia to start.

**Comment by arkid15r** [comment]
From the issue description the process seems to be working as intended. As (I assume) the user isn't a part of our AWS account the authenticated requests don't work. And the publicly accessible bucket content should be available with the `--no-sign-request` option.

Adding the `Authenticated users group (anyone with an AWS account)` could be an alternative (needs to be tested with/without the `Everyone (public access)` ACL enabled).

**Comment by jashapiro** [comment]
Correct, the user is not part of our AWS account, but I had assumed (perhaps incorrectly) that allow everyone access would include authenticated users.
But just in case I did add the  `Authenticated users group (anyone with an AWS account)` option, and I tested it with a profile of my own (outside ALSF) and it seemed to work as expected (I could list the bucket+prefix). So I am still confused as to why this user is getting `AccessDenied`

**Comment by jashapiro** [comment]
I wonder if it is possible to pass to nextflow an option to use a NULL AWS profile somehow? Somehow the profile being used is blocking access, but we would not want to override AWS profiles by default...

**Comment by jashapiro** [comment]
Nextflow has a setting that we might be able to use for anonymous S3 access:
`aws.client.anonymous = true` 

I have asked for testing of this, and we will see if it works

**Comment by jashapiro** [comment]
> Nextflow has a setting that we might be able to use for anonymous S3 access: `aws.client.anonymous = true`
> 
> I have asked for testing of this, and we will see if it works

This was tested successfully! Updating issue to reflect that this should be added to the example config file.

**Comment by jashapiro** [comment]
closed by #206
